db/migrate/20211201224317_migrate_date_produced_in_batches.rb

```diff
@@ -0,0 +1,22 @@
+class MigrateDateProducedInBatches < ActiveRecord::Migration
+  def up
+    Batch.find_each do |batch|
+      next if batch.date_produced.nil?
+      batch.date_produced = if batch.date_produced.is_a?(String)
+                              Time.strptime(value, Batch.date_format[:pattern]) rescue batch.date_produced
+                            else
+                              batch.date_produced
+                            end
+      batch.save
+    end
+  end
+
+  def down
+    # only udpate physical column in the database to not overwrite the value stored in core_fields
+    connection.exec_update(<<-EOQ, "SQL", [[nil,'']])
+      UPDATE  batches
+      SET    date_produced = NULL
+    EOQ
```

Comment:
I had to use plain SQL query since using `update_all` updates the model attribute, causing this to also update the attribute in the `core_fields` field. Updating this value would have cause data loss. I needed to execute plain SQL query so I didn't update the field through `ActiveRecord`

Reply:
Yes, I understood that, but ActiveRecord can be used to write safe SQL requests, which applies to the [.update_all()](https://api.rubyonrails.org/v4.2/classes/ActiveRecord/Relation.html#method-i-update_all) class method:

> This method constructs a single SQL UPDATE statement and sends it straight to the database. It does not instantiate the involved models and it does not trigger Active Record callbacks or validations. Values passed to `update_all` will not go through ActiveRecord's type-casting behavior. It should receive only values that can be passed as-is to the SQL database.

Meaning that `Batch.update_all(date_produced: nil)` will merely generate and execute an `UPDATE batches SET date_produced = NULL` SQL request.